### PR TITLE
feat[added homepage]

### PR DESCRIPTION
**Pull request by Radheshyam1234**
added Homepage with sidebar and product listing
Front-tube-homepage

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,5 +1,25 @@
-import React from "react";
+import React, { useEffect } from "react";
+import { Routes, Route } from "react-router-dom";
+import { useVideosDataProvider } from "./Context/VideosDataContext/VideosDataProvider";
+
+import { Home, Navbar } from "./Components";
+
+import { getVideos } from "./utilities/backendRequest";
+
+import "./styles.css";
```
Comment (shuklahimanshu81) [inline src/App.js]: I will suggest removing extra space

File: src/Components/Home/Header/SearchBox.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React, { useState, useEffect, useCallback } from "react";
+
```
Comment (shuklahimanshu81) [inline src/Components/Home/Header/SearchBox.js:2]: Why did you imported those hook in this component?

File: src/Components/Home/Home.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React, { useEffect } from "react";
+import { Sidebar } from "./Sidebar/Sidebar";
+import { Navbar } from "./Header";
+import { useVideosDataProvider } from "../../Context/VideosDataContext/VideosDataProvider";
+import "./home.css";
+import { VideoCard } from "./VideoCard/VideoCard";
+
+export const Home = () => {
+  useEffect(() => {
+    console.log(videosList);
```
Comment (shuklahimanshu81) [inline src/Components/Home/Home.js]: Remove console log statements.

File: src/Components/Home/Sidebar/sidebar.css [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  display: block;
+  /* font-weight: 500; */
+  padding-top: 5px;
+}
+
+.sidebar-child span i {
+  padding: 10px;
+  margin-right: 10px;
+  /* font-size: 18px; */
+}
```
Comment (shuklahimanshu81) [inline src/Components/Home/Sidebar/sidebar.css]: Remove those unnecessary font-size comments

File: src/utilities/backendRequest.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+export const getVideos = async (setVideosList) => {
+  try {
+    const {
+      data: { videos },
+    } = await axios.get("/api/videos");
+    setVideosList(videos);
+  } catch (error) {
+    alert(error);
+  }
```
Comment (shuklahimanshu81) [inline src/utilities/backendRequest.js]: IMO alerts are not a good way to show errros.

**Review by shuklahimanshu81 — APPROVED** [review]
It would be good if you could answer the question that I have asked. I have suggested the changes like removing comments and console log statements. 
Code structure is clean which I really need to learn from you.

File: src/App.js (hunk 1 above)
Comment (junaidshaikh-js) [reply src/App.js]: I would say batch your imports into logical sections 

Like 

import from the dependencies 
--space--
import from the src files ( components and other function) 
--space--
import CSS files

**Review by junaidshaikh-js — APPROVED** [review]
Good work! 

Clean folder structure and code too. 

Loved the UI

File: src/Components/Home/home.css [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+}
+.chips-container {
+  display: flex;
+}
+.video-description {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+/* .video-title {
```
Comment (Bharatrathod-81) [inline src/Components/Home/home.css]: comment can be removed

**Review by Bharatrathod-81 — APPROVED** [review]

**Pull request merged**